### DISCUSSION TITLE
[探索] 关于 Git 历史数据可视化的工具推荐

### DISCUSSION THREAD

**Discussion by xgdyp** [X-lab2017/open-wonderland#32] [2️⃣ 聊项目]
- git-history
在线服务： 在github上打开一个文件 然后把github.com 改成github.githistory.xyz

- git-extra
一系列git的shell脚本

**Comment by tyn1998** [comment]
[Gource](https://github.com/acaudwell/Gource)必须有排面！

Arch Linux pacman或Mac homebrew都可以直接安装~

**Reply by will-ww** [reply]
每一次 Git contribution 都是值得记录与展示的，都是对全球数字基础设施的一点贡献~

**Reply by tyn1998** [reply]
GitHub在2021年5月支持了[上传视频](https://github.blog/2021-05-13-video-uploads-available-github/)，我来试试~

https://user-images.githubusercontent.com/32434520/168522687-14bff68b-e2df-4a2a-b7f2-cd8c99b818ce.mov

导出视频使用的Gource命令为：
```bash
gource -1000x720 -s 0.2 --key --date-format "%m/%d/%Y, %H:%M:%S" --font-size 40 --user-font-size 20 --title "Visualization of The Git Repo of OpenResearch by Gource" --stop-at-end --logo logo-v.png -o - | ffmpeg -y -r 30 -f image2pipe -vcodec ppm -i - -vcodec libx264 -preset ultrafast -pix_fmt yuv420p -crf 1 -threads 0 -bf 0 gource.mp4
```
PS：原始视频558M，经过软件压缩得到68M的视频；在Retina显示屏下1000x720等价于2000x1440。

Bonus：`gource --output-custom-log my-project-custom.log`命令可以直接导出经过gource分析仓库得到的日志，有了这个日志就不需要仓库中的`.git`目录了，直接使用`gource my-cvs-project.log`就可以播放了！这意味着，如果世界上在未来出现了一个叫gource.js的东西，后端只要事先把`xxx.log`导出存放到oss上，前端的gource.js就可以直接下载小巧的`xxx.log`文件就能实现播放了！

**Reply by tyn1998** [reply]
视频链接上下方**都要空一行**，才能被渲染出来；否则只能看到一个链接。

**Reply by will-ww** [reply]
这个东西，相当优秀呀~

**Reply by will-ww** [reply]
还支持倍速和下载功能~

**Reply by tyn1998** [reply]
嗯~但是有限制：
- 格式：.mp4或.mov
- 大小：100MB

**Reply by AliceCodeZhang** [reply]
视频好酷，是通过各个参与者的history 日志分析出来的吗？ 是否可以有个整体视频描述和题目

**Comment by will-ww** [comment]
> * git-history
>   在线服务： 在github上打开一个文件 然后把github.com 改成github.githistory.xyz
> * git-extra
>   一系列git的shell脚本

另外，你整个标题的名称起的不够好，无法吸引人点入进来~ :-)

我将题目修改了下~

**Reply by xgdyp** [reply]
果然还是王老师起的好哈哈

**Comment by bifenglin** [comment]
推荐图数据可视化工具Gephi

**Reply by xgdyp** [reply]
这个是要边信息进去嘛
